src/edu/stanford/nlp/sempre/interactive/VegaFn.java

```diff
@@ -67,32 +69,44 @@ public DerivationStream call(final Example ex, final Callable c) {
 
   // takes a token determine what kind of value it can take
   static class VegaValueStream extends MultipleDerivationStream {
+    private String curString;
+    private Iterator<String> strIterator;
     private Set<String> types;
     private Iterator<String> typeIterator;
     private Callable callable;
+    private Function<String, Set<String>> tokenChecker;
 
     public VegaValueStream(Example ex, Callable c, Function<String, Set<String>> tokenChecker) {
+      List<String> strings = new ArrayList<String>();
       String child = c.childStringValue(0);
-      types = tokenChecker.apply(child);
+      strings.add(child);
+      if (child.matches(".*[s?!.,]"))
```

Comment:
lets handle in tokenizer

Reply:
Not sure tokenizer is right place, the code is written so that we try both stripping and not stripping the "s".  Adding it in tokenizer would force us to commit to one or the other (and I definitely don't want to rule out the possibility of values that end in "s" or punctuation, e.g. axis titles)

Reply:
Hmm maybe this should be on the lemmatizer, and we should always try the raw token as well as the lemma

Reply:
sounds reasonable. i just thought punctuation is usually handled in tokenizer, and s in lemmatization/normalization. but agree with you that we do not commit to one or the other right now, when its not too clear what role punctuations are gonna play in this system